lib/nx_crc.c

```diff
@@ -261,7 +261,7 @@ local unsigned long nx_crc32_z(crc, buf, len)
 unsigned long ZEXPORT nx_crc32(crc, buf, len)
     unsigned long crc;
     const unsigned char FAR *buf;
-    uint64_t len;
+    unsigned int len;
```

Comment:
This function is part of the API, we have to be very careful before changing it.
Why are we using `unsigned int` if other functions use `unsigned long` for the same parameter?

Reply:
@mscastanho based on your explanation, zlib made a mistake and corrected it later with the `*_z` functions.
IMHO, we don't need to make the same mistake for the `nx_*` functions.
However, it's very important that the functions exported by zlib do have the same declaration and definition.

Besides that, I think we can fix the length for `nx_adler32`. Luckily, a 64-bit unsigned integer is binary compatible with a 32-bit unsigned integer.

Reply:
Furthermore: the `nx_*` API is ours to choose.  We can create more functions as we wish.

Reply:
@tuliom I changed both `nx_adler32` and `nx_crc32` to use `size_t` for the len parameter. So now they are basically equivalent to their `_z` counterparts`.